elements/includes/creative_commons.inc

```diff
@@ -72,269 +124,941 @@ function xml_form_elements_creative_commons($element, &$form_state) {
+ * @param array $values
+ *   The associative value array that was used in the query
+ *   (@see xml_form_elements_creative_commons_make_value_array())
+ * @param SimpleXMLElement $response
+ *   The API response.
+ * @param array $form_state
+ *   The form state.
+ */
+function xml_form_elements_creative_commons_store_api_response($element_name, $storage_folder, $values, $response, &$form_state) {
+  if (!isset($form_state['storage']['xml_form_elements'][$element_name][$storage_folder])) {
+    $form_state['storage']['xml_form_elements'][$element_name][$storage_folder] = array();
+  }
+  $form_state['storage']['xml_form_elements'][$element_name][$storage_folder]['query'] = $values;
+  $form_state['storage']['xml_form_elements'][$element_name][$storage_folder]['response'] = ($response !== FALSE) ? $response->asXML() : FALSE;
+}
+
+/**
+ * Perform API request and cache in storage.
+ *
+ * @param string $element_name
+ *   The `creative_commons` element's name.
+ * @param string $storage_folder
+ *   The storage "subfolder".
+ * @param array $values
+ *   The associative value array to use in the query
+ *   (@see xml_form_elements_creative_commons_make_value_array()).
+ * @param array $form_state
+ *   The form state.
+ *
+ * @return mixed
+ *   SimpleXMLElement the return from the REST API.
+ *   FALSE if failed request.
+ */
+function xml_form_elements_creative_commons_retrieve_api_response($element_name, $storage_folder, $values, &$form_state) {
+  if (isset($form_state['storage']['xml_form_elements'][$element_name][$storage_folder]['query']) &&
+    $form_state['storage']['xml_form_elements'][$element_name][$storage_folder]['query'] == $values &&
+    isset($form_state['storage']['xml_form_elements'][$element_name][$storage_folder]['response']) &&
+    ($form_state['storage']['xml_form_elements'][$element_name][$storage_folder]['response'] !== FALSE)) {
+    // Get response from storage.
+    $response = simplexml_load_string($form_state['storage']['xml_form_elements'][$element_name][$storage_folder]['response'], 'SimpleXMLElement');
+  }
+  else {
+    if (empty($values['license_fieldset']['license_uri'])) {
+      // Without specified URI, get a fresh license based on input values.
+      $api_path = xml_form_elements_creative_commons_get_api_path($values['license_fieldset']['license_type']);
+      // Default to standard for unknown licenses.
+      if (!$api_path) {
+        $api_path = 'license/standard/get';
+      }
+      // Prepare query options.
+      $api_options = array();
+      if ($values['license_fieldset']['license_type'] == 'by') {
+        $api_options = array(
+          'commercial' => $values['license_fieldset']['allow_commercial'],
+          'derivatives' => $values['license_fieldset']['allow_derivatives'],
+          'jurisdiction' => $values['license_fieldset']['license_jurisdiction'],
+        );
+      }
+      $response = xml_form_elements_creative_commons_query_api_get($api_path, $api_options);
```

Comment:
@DiegoPino True, so I moved the common code in a helper function. I left both functions since I though they could be useful on their own. If not, I can reduce to one...